src/metabase/upload.clj

```diff
@@ -529,14 +536,22 @@
                                 :name                   (humanization/name->human-readable-name filename-prefix)
                                 :visualization_settings {}}
                                @api/*current-user*)
-            upload-seconds    (/ (- (System/currentTimeMillis) start-time)
-                                 1000.0)]
+            upload-seconds    (/ (check-timer-ms timer) 1e3)
+            stats             (assoc stats :upload-seconds upload-seconds)]
+
+        (events/publish-event! :event/upload-create
+                               {:user-id  (:id @api/*current-user*)
+                                :model-id (:id table)
+                                :model    :model/Table
+                                :details  {:db-id       db-id
+                                           :schema-name schema-name
+                                           :table-name  table-name
+                                           :model-id    (:id card)
+                                           :stats       stats}})
+
         (snowplow/track-event! ::snowplow/csv-upload-successful
                                api/*current-user-id*
-                               (merge
-                                {:model-id       (:id card)
-                                 :upload-seconds upload-seconds}
```

Comment:
I don't know why we had this backwards merge, neither of those keys are ever in the stats map.